modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceHandler.java

```diff
@@ -130,6 +135,45 @@ public void requestReceived(NHttpServerConnection conn) {
                 return;
             }
 
+
+
+            //check correlationEnabling parameter
+            String correlation_status = System.getProperty(PassThroughConstants.CORRELATION_LOGS_SYS_PROPERTY);
```

Comment:
Reading from a system property is an expensive process. So let's make this a one time process.